cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/index.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+
+import AbstractMultiRowWidget, {
+  IMultiRowProps,
+  IMultiRowWidgetProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget';
+import { IStageSchema } from 'components/AbstractWidget';
+import KeyValueDropdownRow, {
+  IDropdownOption,
+  OrderingEnum,
+} from 'components/AbstractWidget/KeyValueDropdownWidget/KeyValueDropdownRow';
+
+import ThemeWrapper from 'components/ThemeWrapper';
+import { WIDGET_PROPTYPES } from 'components/AbstractWidget/constants';
+import { objectQuery } from 'services/helpers';
+
+interface IInputKeyValueDropdownWidgetProps extends IMultiRowWidgetProps {
+  'key-placeholder'?: string;
+  'kv-delimiter'?: string;
+  dropdownOptions: IDropdownOption[];
+  delimiter?: string;
+  showDelimiter?: boolean;
+  extraConfig?: any;
+}
+
+interface IInputKeyValueDropdownProps extends IMultiRowProps<IInputKeyValueDropdownWidgetProps> {}
+
+interface IField {
+  name: string;
+  type: string[];
+}
+
+class InputFieldDropdownKeyValue extends AbstractMultiRowWidget<IInputKeyValueDropdownProps> {
+  public getFields(schemas: IStageSchema[], allowedTypes: string[]) {
+    let fields = [];
+    if (!schemas || schemas.length === 0) {
+      return fields;
+    }
+    const stage = schemas[0];
+
+    try {
+      const unparsedFields = JSON.parse(stage.schema).fields;
+
+      if (unparsedFields.length > 0) {
+        fields = unparsedFields
+          .filter((field: IField) => this.containsType(field.type, allowedTypes))
+          .map((field: IField) => field.name);
+      }
+    } catch {
+      // tslint:disable-next-line: no-console
+      console.log('Error: Invalid JSON schema');
+    }
+    return fields;
+  }
+
+  /*   Function that checks if types contains a type that is in allowedTypes
```

Comment:
If we need this function please format the multiline comment as,

```
/**
 *  Function that checks if types contains a type that is in allowedTypes
 * This is meant to handle nullable fields since a nullable string type is
 * presented as ['string','null'].
 */
```